tests/v2/validation/provisioning/rke2/ace_test.go

```diff
@@ -0,0 +1,217 @@
+					machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+					machineConfigResp, err := testSessionClient.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+					require.NoError(r.T(), err)
+
+					machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
+
+					cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
+
+					cluster.Spec.LocalClusterAuthEndpoint = r.localClusterAuthEndpoint
+
+					clusterResp, err := clusters.CreateK3SRKE2Cluster(testSessionClient, cluster)
+					require.NoError(r.T(), err)
+
+					kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+					require.NoError(r.T(), err)
+
+					result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+						FieldSelector:  "metadata.name=" + clusterName,
+						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+					})
+					require.NoError(r.T(), err)
+
+					checkFunc := clusters.IsProvisioningClusterReady
+
+					err = wait.WatchWait(result, checkFunc)
+					assert.NoError(r.T(), err)
+					assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+
+					clusterInfo, err := kubeProvisioningClient.Clusters(namespace).Get(context.TODO(), clusterResp.Name, metav1.GetOptions{})
+					require.NoError(r.T(), err)
+
+					clusterID := clusterInfo.Status.ClusterName
+					kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID) //client or r.client
+					require.NoError(r.T(), err)
+
+					original, err := r.client.SwitchContext(clusterResp.Name, kubeConfig)
+					require.NoError(r.T(), err)
+
+					originalResp, err := original.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+					require.NoError(r.T(), err)
+					for _, pod := range originalResp.Items {
+						r.T().Logf("Pod %v", pod.GetName())
+					}
+					contexts, err := kubeconfig.GetContexts(kubeConfig)
+					require.NoError(r.T(), err)
+					var contextNames []string
+					for key, _ := range contexts {
+						if strings.Contains(key, "pool") {
+							contextNames = append(contextNames, key)
+						}
+					}
+					for _, contextName := range contextNames {
+						dynamic, err := r.client.SwitchContext(contextName, kubeConfig)
+						require.NoError(r.T(), err)
+						resp, err := dynamic.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+						require.NoError(r.T(), err)
+						r.T().Logf("Switched Context to %v", contextName)
+						for _, pod := range resp.Items {
+							r.T().Logf("Pod %v", pod.GetName())
+						}
+					}
+				})
+			}
+		}
+	}
+}
+
+func (r *RKE2ACETestSuite) TestProvisioningACE() {
+	for _, providerName := range r.providers {
+		provider := CreateProvider(providerName)
+		r.Provisioning_RKE2ClusterACE(provider)
```

Comment:
Also since you are calling this function does it need to be exported?

Reply:
Removed to align with node_driver test